portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,41 +25,49 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> messages;
-  
-  /** TODO: Delete after implementing load from Datastore **/
-  @Override
-  public void init(){
-    messages = new ArrayList<>();
-  }
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJson(messages);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList<>();
+    for(Entity entity : results.asIterable()){
+      comments.add((String) entity.getProperty("comment"));
+    }
+    
+    String json = convertToJson(comments);
+
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = request.getParameter("comment");
+    long timestamp = System.currentTimeMillis();
     
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("comment", comment);
```

Comment:
Totally optional topic you can research if you're feeling like an extra challenge: I think Datastore might have a limit for how long a String can be?

(Like I said, this is totally optional! Feel free to leave this for later or to disregard it entirely.)